libs/bulldozer/application/features/code-editor/src/lib/code-editor.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, Input } from '@angular/core';
+import { DarkThemeService } from '@heavy-duty/bulldozer/application/ui/dark-theme';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'bd-code-editor',
+  template: `
+    <ngx-monaco-editor
+      class="{{ customClass }}"
+      [options]="editorOptions$ | ngrxPush"
+      [ngModel]="template"
+    ></ngx-monaco-editor>
+  `,
+  styles: [],
+})
+export class CodeEditorComponent {
+  @Input() template: string | null = null;
+  @Input() customClass = '';
+  @Input() readOnly = false;
+
+  readonly isDarkThemeEnabled$ = this._themeService.isDarkThemeEnabled$;
+  readonly editorOptions$ = this._themeService.isDarkThemeEnabled$.pipe(
```

Comment:
Is there a reason for doing this here instead of passing it as an Input property? By doing that, you would remove the ThemeService dependency